File: language-support/js/daml-ledger/src/validation/completion_stream_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { object, Validation, array, native } from "./base";
import { RequiredValidation } from "./base/typelevel";
import { LedgerOffset } from "./ledger_offset";

function required(): RequiredValidation<ledger.CompletionStreamRequest> {
    return {
        applicationId: native('string'),
        offset: LedgerOffset,
        parties: array(native('string')),
    };
}

export const CompletionStreamRequest: Validation = object<ledger.CompletionStreamRequest>('CompletionStreamRequest', required, () => ({}));

File: canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/messages/command/completion/CompletionStreamRequest.scala
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.ledger.api.messages.command.completion

import com.daml.lf.data.Ref
import com.digitalasset.canton.ledger.api.domain.{LedgerId, ParticipantOffset}

final case class CompletionStreamRequest(
    ledgerId: Option[LedgerId],
    applicationId: Ref.ApplicationId,
    parties: Set[Ref.Party],
    offset: Option[ParticipantOffset],
)
